app/services/api/v1x0/catalog/add_to_order.rb

```diff
@@ -19,7 +19,9 @@ def process
         private
 
         def order_item_params
-          @params.permit(:order_id, :portfolio_item_id, :count, :service_parameters => {}, :provider_control_parameters => {})
+          @params.permit(:order_id, :portfolio_item_id, :count, :service_parameters => {}, :provider_control_parameters => {}).tap do |params|
+            params[:name] = PortfolioItem.find(@params[:portfolio_item_id]).name
```

Comment:
Suggest to use `find_by` here, in case of `find` raise exception. Use a default name for this case?

Reply:
@bzwei Why do we need the name in the OrderItem can't we get that from the relationship portfolio_item_id? What happens if the Job Template name changes in the tower. We are already storing the portfolio_item_id. When creating the internal workspace for doing substitution we are just going to hardcode "product" not its actual name.

Reply:
@mkanoor This is originally a request from @lgalis. Because UI always needs to make an additional call to get the name from portfolio_item. 
When building workspace for substitution the name is not needed by product, but needed for before/after items. We currently use portfolio_item's name to reference before/after. With this name field we now can reference them by order process name which should be preferred.

Order Item is a historical data. We shall not need to maintain its name always to match Job Template (or portfolio items)'s latest name. If my understanding is wrong, i.e., we need to always display the current name even if the order was made years ago, then we should drop this PR.

Reply:
@hsong-rh I think we should let it raise an exception if we cannot find the portfolio item. It makes no sense to continue to the order because it is deemed to fail.

Reply:
> Order Item is a historical data. We shall not need to maintain its name always to match Job Template (or portfolio items)'s latest name. If my understanding is wrong, i.e., we need to always display the current name even if the order was made years ago, then we should drop this PR.

I think you're correct, the order is historical data. If we end up deleting the portfolio item (or discarding, in our case), we should be able to tell the user what portfolio item it was that was ordered at that time.

If we think of John's Amazon examples, if you order a product and the seller changes the product name, I'm not 100% certain that this happens every time but I know I've seen things say "this item no longer exists, but this new version does <here>". So it would be the same kind of thing, you can look at your order history and see what you ordered even if it doesn't necessarily match up with what the catalog contains today.